packages/web-client/tests/unit/layer-1-connection-manager-test.ts

```diff
@@ -0,0 +1,253 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import { ConnectionManager } from '@cardstack/web-client/utils/web3-strategies/layer-1-connection-manager';
+import Web3 from 'web3';
+import Sinon from 'sinon';
+
+module('Unit | layer 1 connection manager', function (hooks) {
+  setupTest(hooks);
+
+  test('It should be able to be instantiated for Kovan and Sokol', async function (assert) {
+    let sokolConnectionManager = new ConnectionManager('sokol');
+    let kovanConnectionManager = new ConnectionManager('kovan');
+
+    assert.ok(
+      sokolConnectionManager,
+      'Sokol connection manager is instantiated'
+    );
+
+    assert.ok(
+      kovanConnectionManager,
+      'Kovan connection manager is instantiated'
+    );
+  });
+
+  test('It should not have a strategy, provider, or providerId at instantiation', async function (assert) {
+    let connectionManager = new ConnectionManager('kovan');
+
+    assert.ok(
+      !connectionManager.strategy,
+      'There is no strategy at first instantiation'
+    );
+    assert.ok(
+      !connectionManager.provider,
+      'There is no provider at first instantiation'
+    );
+    assert.ok(
+      !connectionManager.providerId,
+      'There is no providerId at first instantiation'
+    );
+  });
```

Comment:
I wonder if this is a case of testing the implementation instead of the interface/behaviour? With unit tests there’s a danger of writing assertions that are too tightly coupled to implementation details that aren’t really that important. Often this is the case when assertions are being made on the private properties of the test under class.

Reply:
**providerId**: Used in the UI to decide whether to show MetaMask or WalletConnect-specific images/copy so this is also part of the interface. 

**provider**: I think this is more towards implementation.

**Strategy**: Definitely implementation.

Reply:
I think keeping the test for web3.setProvider is good but we can omit the check for provider and strategy being defined.

Reply:
Dropped assertions for provider and strategy

Reply:
nice 🤩